cdap-api/src/main/java/co/cask/cdap/api/metadata/MetadataWriterContext.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.api.metadata;
+
+import java.util.Map;
+
+/**
+ * The context for emitting metadata from programs
+ */
+public interface MetadataWriterContext {
+
+  /**
+   * Adds the specified {@link Map} to the metadata of the specified {@link MetadataEntity metadataEntity}.
+   * Existing keys will be updated with new values.
+   */
+  void addProperties(MetadataEntity metadataEntity, Map<String, String> properties);
+
+  /**
+   * Adds the specified tags to specified {@link MetadataEntity}.
```

Comment:
In your above code, tags are represented as `Set<String>`, this means duplicated tags will be removed, right? If that's the case, I think it's better for this javadoc to specify that behavior.

Reply:
Since